core/src/test/java/com/graphhopper/routing/util/NameSimilarityEdgeFilterTest.java

```diff
@@ -0,0 +1,227 @@
+        assertTrue(edgeFilter.accept(edge));
+
+        edgeFilter = new NameSimilarityEdgeFilter(new DefaultEdgeFilter(new CarFlagEncoder()), "Johannesstraße, 99636, Rastenberg, Deutschland");
+        edge = new TestEdgeIterator("Laufamholzstraße, ST1333");
+        assertFalse(edgeFilter.accept(edge));
+
+        edge = new TestEdgeIterator("Johannesstraße");
+        assertTrue(edgeFilter.accept(edge));
+
+        edgeFilter = new NameSimilarityEdgeFilter(new DefaultEdgeFilter(new CarFlagEncoder()), "Hauptstraße, 39025, Naturns, Italien");
+        edge = new TestEdgeIterator("Teststraße");
+        assertFalse(edgeFilter.accept(edge));
+
+        edge = new TestEdgeIterator("Hauptstraße");
+        assertTrue(edgeFilter.accept(edge));
+    }
+
+    @Test
+    public void testAcceptWithTypos() {
+        EdgeFilter edgeFilter = new NameSimilarityEdgeFilter(new DefaultEdgeFilter(new CarFlagEncoder()), "Laufamholzstraße 154 Nürnberg");
+        EdgeIteratorState edge = new TestEdgeIterator("Laufamholzstraße, ST1333");
+        assertTrue(edgeFilter.accept(edge));
+
+        // Single Typo
+        edgeFilter = new NameSimilarityEdgeFilter(new DefaultEdgeFilter(new CarFlagEncoder()), "Kaufamholzstraße 154 Nürnberg");
+        assertTrue(edgeFilter.accept(edge));
+
+        // Two Typos
+        edgeFilter = new NameSimilarityEdgeFilter(new DefaultEdgeFilter(new CarFlagEncoder()), "Kaufamholystraße 154 Nürnberg");
+        assertTrue(edgeFilter.accept(edge));
+
+        // Three Typos
+        edgeFilter = new NameSimilarityEdgeFilter(new DefaultEdgeFilter(new CarFlagEncoder()), "Kaufmholystraße 154 Nürnberg");
+        assertFalse(edgeFilter.accept(edge));
+
+        edgeFilter = new NameSimilarityEdgeFilter(new DefaultEdgeFilter(new CarFlagEncoder()), "Hauptstraße");
+        edge = new TestEdgeIterator("Hauptstraße");
+        assertTrue(edgeFilter.accept(edge));
+
+        // Single Typo
+        edgeFilter = new NameSimilarityEdgeFilter(new DefaultEdgeFilter(new CarFlagEncoder()), "Hauptstrase");
+        assertTrue(edgeFilter.accept(edge));
+
+        // Two Typos
+        edgeFilter = new NameSimilarityEdgeFilter(new DefaultEdgeFilter(new CarFlagEncoder()), "Hauptstrasi");
+        assertFalse(edgeFilter.accept(edge));
+
+        //TODO Maybe we should not allow matching too short strings here?
+        // Distance - PerfectDistance = 1
+        edgeFilter = new NameSimilarityEdgeFilter(new DefaultEdgeFilter(new CarFlagEncoder()), "z");
+        assertTrue(edgeFilter.accept(edge));
+        // Distance - PerfectDistance = 1
+        edgeFilter = new NameSimilarityEdgeFilter(new DefaultEdgeFilter(new CarFlagEncoder()), "az");
+        assertTrue(edgeFilter.accept(edge));
+
+        // Distance - PerfectDistance = 2
+        edgeFilter = new NameSimilarityEdgeFilter(new DefaultEdgeFilter(new CarFlagEncoder()), "xy");
+        assertFalse(edgeFilter.accept(edge));
+    }
+
+    static class TestEdgeIterator implements EdgeIteratorState {
```

Comment:
Can you use GHUtility.createMockedEdgeIteratorState here?

Oh it looks like there is another helper ShortFastestWeightingTest.createEdge ... will try to fix this.